components/learn/modules/ROOT/pages/on-upgrades.adoc

```diff
@@ -1,8 +1,280 @@
+
+For the sake of the example, let's say we want to add a new feature: a function that increments the `value` stored in the `Box`.
+
+```solidity
+// contracts/Box.sol
+contract Box {
+    // ...
+
+    // Increments the stored value by 1
+    function increment() public {
+        value = value + 1;
+        emit ValueChanged(value);
+    }
+}
+```
+
+After changing the Solidity file, we can now just upgrade the instance we had deployed earlier by running the `openzeppelin upgrade` command.
+
+```bash
+$ npx openzeppelin upgrade
+? Pick a network: development
+✓ Compiled contracts with solc 0.5.9
+✓ Contract Box deployed
+? Which proxies would you like to upgrade?: All proxies
+Instance upgraded at 0xCfEB869F69431e42cdB54A4F4f105C19C080A601.
+```
+
+Done! Our `Box` instance has been upgraded to the latest version of the code, while keeping its state and the same address as before. We haven't had to manually deploy a new one at a new address, nor manually copy the `value` from the old `Box` to the new one.
+
+Let's try it out by invoking the new `increment` function, and checking the `value` afterwards:
+
+```bash
+$ npx openzeppelin send-tx
+? Pick a network: development
+? Pick an instance: Box at 0xCfEB869F69431e42cdB54A4F4f105C19C080A601
+? Select which function: increment()
+✓ Transaction successful: 0x9c84faf32a87a33f517b424518712f1dc5ba0bdac4eae3a67ca80a393c555ece
+Events emitted:
+- ValueChanged(6)
+
+$ npx openzeppelin call
+? Pick a network: development
+? Pick an instance: Box at 0xCfEB869F69431e42cdB54A4F4f105C19C080A601
+? Select which function: retrieve()
+✓ Method 'retrieve()' returned: 6
+```
+
+That's it! Notice how the `value` of the `Box` was preserved throughout the upgrade, as well as its address. And this process is the same regardless of whether you are working on a local blockchain, a testnet, or the main network. Let's see how the OpenZeppelin SDK accomplishes this.
+
+[[how-upgrades-work]]
+== How upgrades work
+
+_This section will be more theory-heavy than others, so feel free to skip over it, and return later if you are curious._
+
+When you create a new upgradeble contract instance, the OpenZeppelin SDK actually deploys two contracts:
+
+. The contract you have written, which is known as the _implementation contract_ or _logic contract_.
+. A _proxy_ to that contract, which is the instance that you actually interact with.
+
+Here, the _proxy_ is a simple contract that just _delegates_ all calls to an implementation contract. A _delegate call_ is similar to a regular call, but all code is executed in the context of the caller, not of the callee. This means that a `transfer` in the implementation contract's code will actually transfer the proxy's balance. Similarly, any reads or writes to the contract storage will read or write from the proxy's storage.
```

Comment:
```suggestion
Here, the _proxy_ is a simple contract that just _delegates_ all calls to an implementation contract. A _delegate call_ is similar to a regular call, except that all code is executed in the context of the caller, not of the callee. Because of this, a `transfer` in the implementation contract's code will actually transfer the proxy's balance, and any reads or writes to the contract storage will read or write from the proxy's own storage.
```